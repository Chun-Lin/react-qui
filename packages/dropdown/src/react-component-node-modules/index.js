export { default as Dropdown } from './components/Dropdown/Dropdown';
