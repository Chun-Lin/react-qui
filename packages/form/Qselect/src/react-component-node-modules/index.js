export { default as Qselect } from './components/Qselect/Qselect';
