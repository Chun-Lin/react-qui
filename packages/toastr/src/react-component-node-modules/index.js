export { default as Toastr } from './components/Toastr/Toastr';
export { default as ToastrContainer } from './components/Toastr/ToastrContainer'
