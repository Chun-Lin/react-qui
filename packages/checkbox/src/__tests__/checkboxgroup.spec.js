import React from 'react'
import ReactDOM from 'react-dom'
import { mount, shallow } from 'enzyme'
import { CheckboxGroup } from '../react-component-node-modules/index'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

it('render <CheckboxGroup> component', () => {
  const checkboxGroup = shallow(
    <CheckboxGroup
      options={['Apple', 'Banana', 'Cat']}
      span={30}
      theme="light"
      direction="row"
    />,
  )
  expect(checkboxGroup).toMatchSnapshot()
})

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <CheckboxGroup
      options={['Apple', 'Banana', 'Cat']}
      span={30}
      theme="light"
      direction="row"
    />,
    div,
  )
})

it('render CheckboxGroup correctly', () => {
  const checkboxGroup = mount(
    <CheckboxGroup
      options={['Apple', 'Banana', 'Cat']}
      span={30}
      theme="light"
      direction="row"
    />,
  )
  expect(checkboxGroup.find('span').length).toEqual(3)
  expect(checkboxGroup.children().text()).toEqual('AppleBananaCat')
})
