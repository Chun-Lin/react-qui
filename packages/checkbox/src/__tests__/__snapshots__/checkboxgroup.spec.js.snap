// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <CheckboxGroup> component 1`] = `
<styled.div>
  <mapProps(withState(withHandlers(Checkbox)))
    checked={false}
    direction="row"
    disabled={false}
    key="Apple"
    span={30}
    theme="light"
  >
    Apple
  </mapProps(withState(withHandlers(Checkbox)))>
  <mapProps(withState(withHandlers(Checkbox)))
    checked={false}
    direction="row"
    disabled={false}
    key="Banana"
    span={30}
    theme="light"
  >
    Banana
  </mapProps(withState(withHandlers(Checkbox)))>
  <mapProps(withState(withHandlers(Checkbox)))
    checked={false}
    direction="row"
    disabled={false}
    key="Cat"
    span={30}
    theme="light"
  >
    Cat
  </mapProps(withState(withHandlers(Checkbox)))>
</styled.div>
`;
